hooks/gen_docs/gen_docs_components.py

```diff
@@ -73,6 +78,29 @@ class KpopsComponent(NamedTuple):
     specific_attrs: list[str]
 
 
+# TODO(Ivan Yordanov): Evaluate whether it makes sense to instead use `__mro__`
+# The problem is that we need an object for that. Note that `__mro__` differs
+# from the output of this function sometimes and `__mro__` probably is the more
+# accurate way.
```

Comment:
Opinions?

Reply:
Difference between `StreamsApp.mro()` and `collect_parents("streams-app"):

<details>

<summary>MRO</summary>

```
In [3]: StreamsApp.mro()
Out[3]:
[kpops.components.streams_bootstrap.streams.streams_app.StreamsApp,
 kpops.components.base_components.kafka_app.KafkaApp,
 kpops.components.streams_bootstrap.StreamsBootstrap,
 kpops.components.base_components.helm_app.HelmApp,
 kpops.components.base_components.kubernetes_app.KubernetesApp,
 kpops.components.base_components.pipeline_component.PipelineComponent,
 kpops.components.base_components.base_defaults_component.BaseDefaultsComponent,
 kpops.utils.pydantic.DescConfigModel,
 pydantic.main.BaseModel,
 abc.ABC,
 object]
```

</details>

<details>

<summary>collect_parents</summary>

```
In [4]: collect_parents("streams-app")
Out[4]:
['streams-app',
 'kafka-app',
 'streams-bootstrap',
 'pipeline-component',
 'helm-app',
 'base-defaults-component',
 'kubernetes-app']

```

</details>